File: src/main/java/no/nav/syfo/kafka/Producer.java
package no.nav.syfo.kafka;

import lombok.extern.slf4j.Slf4j;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;

import static java.util.Collections.singletonMap;
import static java.util.UUID.randomUUID;

@Component
@Slf4j
public class Producer {
    private KafkaTemplate<String, String> kafkaTemplate;

    public Producer(KafkaTemplate<String, String> kafkaTemplate) {
        this.kafkaTemplate = kafkaTemplate;
    }

    public void send() {
        kafkaTemplate.send(
                new SyfoProducerRecord<>(
                        "$topicName$",
                        randomUUID().toString(),
                        "data",
                        singletonMap("example", "header")));
        log.info("Data lagt på kø");
    }
}

File: src/main/kotlin/no/nav/syfo/kafka/Producer.kt
package no.nav.syfo.kafka

import no.nav.syfo.kafka.sykepengesoknad.dto.SykepengesoknadDTO
import no.nav.syfo.log
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Component
import java.util.Collections.singletonMap
import java.util.UUID.randomUUID

@Component
class Producer(private val kafkaTemplate: KafkaTemplate<String, SykepengesoknadDTO>) {
    val log = log()

    fun send() {
        kafkaTemplate.send(
                SyfoProducerRecord(
                        "syfoaltinn-rebehandling-v1",
                        randomUUID().toString(),
                        SykepengesoknadDTO.builder().build(),
                        singletonMap<String, Any>("example", "header")
                )
        )
        log.info("Data lagt på kø")
    }
}
